libraries/cmd/unit_generic.cpp

```diff
@@ -2953,9 +2838,11 @@ bool Unit::RepairUpgradeCargo(Cargo *item, Unit *baseUnit) {
                 if (price <= ComponentsManager::credits) {
                     ComponentsManager::credits -= price;
                     
-                    if (notadditive) {
-                        this->Upgrade(un, 0, 0, 0, true, percentage, makeTemplateUpgrade(this->name, this->faction));
-                    }
+                    // TODO: look closer at this
+                    //if (notadditive) {
+                    //    this->Upgrade(un, 0, 0, 0, true, percentage, makeTemplateUpgrade(this->name, this->faction));
+                    //}
```

Comment:
I think this should be left in, not commented out. It looks to me like, overall, this PR is going to break multiplicative upgrades.

Reply:
I've deprecated both additive and multiplicative upgrades as generalized upgrades several months ago.
Multiplicative exists as a special case for engine_upgrade and afterburner_upgrade.

Looking at what was supported by the game, additive was only supported for SPEC capacitors and cargo expansions. 
Multiplicative was supported for afterburner upgrade and various shady.
Both cargo upgrades (mult_shady_moreupgrade__upgrades and add_cargo_expansion__upgrades) are currently supported. As for the rest, I can open an issue and figure out a way to support them.